opal/mca/event/libevent2022/configure.m4

```diff
@@ -189,9 +180,23 @@ AC_DEFUN([MCA_opal_event_libevent2022_DO_THE_CONFIG], [
     # libevent/include/event2/event-config.h!).  Otherwise, set it to
     # 0.
     libevent_file=$libevent_basedir/libevent/config.h
+
+    # If we are not building the internal hwloc, then indicate that
+    # this component should not be built.  NOTE: we still did all the
+    # above configury so that all the proper GNU Autotools
+    # infrastructure is setup properly (e.g., w.r.t. SUBDIRS=libevent in
+    # this directory's Makefile.am, we still need the Autotools "make
+    # distclean" infrastructure to work properly).
+
+    if test "$with_libevent" != "internal" && test -n "$with_libevent" && test "$with_libevent" != "yes"; then
+        libevent_happy=no
```

Comment:
It might be useful to put some kind of output in here to indicate *why* we're not going to build this component.  For example, perhaps something like this:

```sh
AC_MSG_CHECKING([if external libevent is being used])
AC_MSG_RESULT([yes (this component will be skipped)])
```